models/evc.py

```diff
@@ -1058,32 +1080,128 @@ def run_sdntrace(uni):
+        circuits_checked[circuit_id] = True
+        trace_a = circuit_by_traces[circuit_id]['trace_a']
+        trace_z = circuit_by_traces[circuit_id]['trace_z']
+        if len(trace_a) != len(circuit_current_path) + 1:
             log.warning(f"Invalid trace from uni_a: {trace_a}")
-            return False
-        trace_z = self.run_sdntrace(self.uni_z)
-        if len(trace_z) != len(self.current_path) + 1:
+            circuits_checked[circuit_id] = False
+        if len(trace_z) != len(circuit_current_path) + 1:
             log.warning(f"Invalid trace from uni_z: {trace_z}")
-            return False
-
-        for link, trace1, trace2 in zip(self.current_path,
+            circuits_checked[circuit_id] = False
+        for link, trace1, trace2 in zip(circuit_current_path,
                                         trace_a[1:],
                                         trace_z[:0:-1]):
+            metadata_vlan = None
+            if link.metadata:
+                metadata_vlan = glom(link.metadata, 's_vlan.value')
             if compare_endpoint_trace(
-               link.endpoint_a,
-               glom(link.metadata, 's_vlan.value'), trace2) is False:
+                                        link.endpoint_a,
+                                        metadata_vlan,
+                                        trace2
+                                    ) is False:
                 log.warning(f"Invalid trace from uni_a: {trace_a}")
-                return False
+                circuits_checked[circuit_id] = False
             if compare_endpoint_trace(
-               link.endpoint_b,
-               glom(link.metadata, 's_vlan.value'), trace1) is False:
+                                        link.endpoint_b,
+                                        metadata_vlan,
+                                        trace1
+                                    ) is False:
                 log.warning(f"Invalid trace from uni_z: {trace_z}")
-                return False
+                circuits_checked[circuit_id] = False
 
-        return True
+    @staticmethod
+    # pylint: disable=too-many-locals
+    def check_list_traces(list_circuits):
+        """Check if current_path is deployed comparing with SDN traces."""
+        if not list_circuits:
+            return {}
+        uni_list = []
+        circuit_data = {}
+        for circuit_id in list_circuits:
+            circuit = list_circuits[circuit_id]
+            uni_list.append(circuit.uni_a)
+            uni_list.append(circuit.uni_z)
+            dpid_a = circuit.uni_a.interface.switch.dpid
+            port_a = circuit.uni_a.interface.port_number
+            interface_a = str(dpid_a) + ':' + str(port_a)
+            circuit_data[interface_a] = {
+                                            'circuit_id': circuit.id,
+                                            'trace_name': 'trace_a'
+                                        }
```

Comment:
Gretel, since we can have multiple EVCs in the same UNI, indexing the circuit_data by the interface_id is not secure. A better way would be to include the VLAN (if there is a VLAN in the tag -- please be prepared for some EVCs which dont have VLAN, they are called EPL - Ethernet Private Line - in the mef_eline terminology). If we dont include the VLAN we may end up getting the wrong data once the traces return (Line 1185).